src/main.py

```diff
@@ -1,17 +1,22 @@
-import atexit
 import logging
 import tkinter as tk
 from tkinter import Button, Canvas, Checkbutton, Entry, IntVar, Scrollbar, messagebox
 
 import globals
+import argparse 
+parser = argparse.ArgumentParser() 
+parser.add_argument('-d','--debug', action='store_true', default=False) 
+args = parser.parse_args() 
+globals.debug_mode |= args.debug
+
 from canvas_operations import (
     enter_state,
     maximize_visible_canvas,
     on_canvas_click,
     on_canvas_scroll,
     render_uml_diagram,
     show_hints,
-    zoom,
+    zoom,   
```

Comment:
I was expecting the linter to complain about this trailing whitespace.

In any case, it is probably a good idea to make the provided `.editorconfig` file apply to all your edits, by tuning your editor, based on https://editorconfig.org/#pre-installed or https://editorconfig.org/#download.